test/lit.cfg

```diff
@@ -334,6 +334,7 @@ config.scale_test = make_path(config.swift_utils, 'scale-test')
 config.PathSanitizingFileCheck = make_path(config.swift_utils, 'PathSanitizingFileCheck')
 config.swift_lib_dir = make_path(config.swift, '..', '..', 'lib')
 config.round_trip_syntax_test = make_path(config.swift_utils, 'round-trip-syntax-test')
+config.refactor_check_compiles = make_path(config.swift_utils, 'refactor-check-compiles.py')
```

Comment:
FWIW this refactoring will often fail to compile (ie. placeholders, removing ?, etc). I like this idea in theory, but it's going to fail for many of the existing calls, eg. the convert call refactors won't compile. I suppose we could add the async variants in that case 🤔. Seems like a decent idea in general though.

Reply:
My idea was that we can use it in places where we know that the refactored code should compile. In others we can still use the plain `%refactor`. Just another tool in the toolbox and I found it very handy while developing this PR.